Makefile

```diff
@@ -0,0 +1,5 @@
+build:
+	sed -e "s/\$${UNISON_VERSION}/2.48.3/"   -e "s/\$${OCAML_VERSION}/4.02/" Dockerfile > Dockerfile-unison2.48.3-ocaml4.02
+	sed -e "s/\$${UNISON_VERSION}/2.48.3/"   -e "s/\$${OCAML_VERSION}/4.01/" Dockerfile > Dockerfile-unison2.48.3-ocaml4.01
+	sed -e "s/\$${UNISON_VERSION}/2.40.102/" -e "s/\$${OCAML_VERSION}/4.02/" Dockerfile > Dockerfile-unison2.40.102-ocaml4.02
+	sed -e "s/\$${UNISON_VERSION}/2.40.102/" -e "s/\$${OCAML_VERSION}/4.01/" Dockerfile > Dockerfile-unison2.40.102-ocaml4.01
```

Comment:
It might be cleaner to enlist the help of an array here:
```bash
build:
	declare -a UNISON_VERSIONS=("2.48.3" "2.40.102") \
	declare -a OCAML_VERSIONS=("4.02" "4.01") \
	for uv in "${UNISON_VERSIONS[@]}"; do \
		for ov in "${OCAML_VERSIONS[@]}"; do \
			sed -e "s/^(ARG\sUNISON_VERSION=)(([0-9]{1}).([0-9]{2}.([0-9]{1})))/${uv}/"   -e "s/^(ARG\sUNISON_VERSION=)(([0-9]{1}).([0-9]{2}))/${ov}/" Dockerfile > Dockerfile-unison${uv}-ocaml${uv}; \
		done; \
	done; 
```
* Note, this is largely untested...